src/Server/TCPHandler.cpp

```diff
@@ -293,6 +295,12 @@ void TCPHandler::runImpl()
             if (e.code() == ErrorCodes::UNKNOWN_PACKET_FROM_CLIENT)
                 throw;
 
+            /// If there is UNEXPECTED_PACKET_FROM_CLIENT emulate network_error
+            /// to break the loop, but do not throw to send the exception to
+            /// the client.
+            if (e.code() == ErrorCodes::UNEXPECTED_PACKET_FROM_CLIENT)
+                network_error = true;
+
             /// If a timeout occurred, try to inform client about it and close the session
             if (e.code() == ErrorCodes::SOCKET_TIMEOUT)
                 network_error = true;
```

Comment:
The INSERT is often followed by data which should be inserted, and those data aren't involved in this hash calculation. Not sure if we should do something about it right now though.

Reply:
Can you elaborate?

Reply:
When `node1` sends an INSERT query to `node2` and some bad guy is in the middle of that communication, he can't change the initial user or the query itself, but he can change the data that will be inserted into `node2` because the data are not protected with the hash.

Reply:
Oh I see, thank you (I though that this was related to that error handling). To protect from this the hash of the INSERTed blocks should be added, but it will heart performance (a lot).

Reply:
Actually your can and should use ssl to protect from this use case